frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileWriter.java

```diff
@@ -0,0 +1,64 @@
+package com.virtuslab.gitmachete.frontend.file;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.intellij.openapi.application.ReadAction;
+import com.intellij.openapi.vfs.VfsUtilCore;
+import com.intellij.openapi.vfs.VirtualFileManager;
+import io.vavr.collection.List;
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import lombok.CustomLog;
+import lombok.val;
+
+import com.virtuslab.branchlayout.api.BranchLayout;
+import com.virtuslab.branchlayout.api.BranchLayoutException;
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.branchlayout.impl.readwrite.BranchLayoutFileUtils;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@CustomLog
+public class MacheteFileWriter {
+
+  @UIThreadUnsafe
+  @SuppressWarnings("regexp") // to allow for `synchronized`
+  public synchronized void writeBranchLayout(Path path, BranchLayout branchLayout, boolean backupOldFile,
```

Comment:
it can be static, right?

Reply:
I wanted to make it static, but then there's this argument in some methods connected to the VFS:
<img width="559" alt="image" src="https://user-images.githubusercontent.com/61919032/199475300-06dca822-eb8d-48dc-9d45-4f5f551afd63.png">
And here:
<img width="374" alt="image" src="https://user-images.githubusercontent.com/61919032/199475513-93e2648c-7a23-41d5-9d9a-90dab84b8171.png">

So we need some kind of object reference.

Looks like this requestor information could then be used when reacting to VirtualFile changes using some listener e.g. like in EnhancedGraphTable. As we don't use it I guess it could be null. What do you think?

Reply:
Okay, both methods will be static, but they require one more argument for the sake of using this `requestor` mechanism correctly. WDYT?

Reply:
Hmmm
>  I guess it could be null. What do you think?

but
>  they will require one more argument for the sake of using this requestor mechanism correctly

So after all, the `requestor` canNOT be `null` in this case? 🤔

Reply:
Well it CAN be for sure, if we never plan on using any more sophisticated VFS changes listeners that's totally fine (if IntelliJ doesn't use this field in any other way but that's hard to tell).
I just thought that putting a new argument there won't be a problem but we use the `requestor` mechanism as intended.

Reply:
Okay, won't harm to pass `requestor` as argument probably